### DISCUSSION TITLE
[Closed] Absolute layout LayoutBounds and LayoutFlag extension

### DISCUSSION THREAD

**Discussion by danielftz** [CommunityToolkit/Maui.Markup#22] [New Feature Discussions]
I propose a ViewInAbsoluteLayoutExtensions static class, where LayoutBounds and LayoutFlag static methods (and their overlaods) are implemented. I have already coded these up in my own project, just thought you could be useful being in the package itself.

**Comment by pictos** [comment]
@danielftz thanks for the proposal. Could you provide the API spec for this feature? That will help us to make a decision.

**Reply by danielftz** [reply]
MCT markup is currently missing helper extension methods for Absolute layout attached properties, specifically LayoutBounds and LayoutFlags. 

### API

```
public static TView LayoutFlags<TView>(this TView view, AbsoluteLayoutFlags flag) where TView : View
		
public static TView LayoutBounds<TView>(this TView view, int x, int y) where TView : View
		
public static TView LayoutBounds<TView>(this TView view, Point point) where TView : View

public static TView LayoutBounds<TView,TCoord>(this TView view, TCoord x, TCoord y) where TView : View where TCoord : Enum
		
public static TView LayoutBounds<TView>(this TView view, int x, int y, int width , int height) where TView : View

public static TView LayoutBounds<TView, TCoord>(this TView view, TCoord x, TCoord y, TCoord width, TCoord height) where TView : View where TCoord : Enum

public static TView LayoutBounds<TView>(this TView view, Point point, Size size) where TView : View
		
public static TView LayoutBounds<TView>(this TView view, Rectangle rectangle) where TView : View

```

###  Scenario

```
BoxView label = new BoxView
{
}.LayoutBounds(0, 0, 10, 10)
.LayoutFlags(AbsoluteLayoutFlags.None);
```


### Difficulty
Low

Suggestions are very welcome. I think there is a possibility that there are too many overloads...

**Comment by danielftz** [comment]
Will I be assigned a pull request? Or do I make a pull request myself? Or how does this work?

**Reply by pictos** [reply]
We need to discuss and vote on this idea first.
cc: @brminnick

**Reply by TheCodeTraveler** [reply]
Thanks @danielftz! Could you open a [New Feature Proposal](https://github.com/CommunityToolkit/Maui.Markup/issues/new?assignees=&labels=new%2C+proposal&template=open-a-new-feature-proposal.md&title=%5BProposal%5D+), and we'll vote to approve it at our next Standup.

**Comment by CliffAgius** [comment]
Closed and answered
